### PR TITLE
Generalize election module to support items instead of participants, and add the initial gift card exchange game.

### PR DESCRIPTION
**Pull request by cjqian**

### CODE REVIEW

File: frontend/src/components/stages/election_editor.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    `;
+  }
+
+  private renderElectionItems() {
+    if (!this.stage) return nothing;
+    const electionItems: ElectionItem[] = this.stage.electionItems || [];
+
+    const addItem = () => {
+      if (!this.stage) return;
+      const newId = String(electionItems.length + 1); // Generate a numeric ID as string
```
Comment (vivtsai) [inline frontend/src/components/stages/election_editor.ts:74]: Can we use `generateId` from `utils/shared` or some other random method? This is going to generate duplicates if an item is deleted

Comment (vivtsai) [reply frontend/src/components/stages/election_editor.ts:74]: could also add a `createElectionItem` function in `utils/stages/election_stage.ts`

File: frontend/src/components/stages/election_view.ts [hunk 2]
```diff
@@ -94,18 +121,25 @@ export class ElectionView extends MobxLitElement {
...
   }
 
-  private renderDraggableParticipant(profile: ParticipantProfile) {
-    if (profile.publicId === this.participantService.profile?.publicId) {
-      return nothing;
-    }
+  private renderElectionItem(item: ElectionItem) {
+    // TODO: Allow item photos?
+    return html`
+        <div class="right">
```
Comment (vivtsai) [inline frontend/src/components/stages/election_view.ts:127]: Missing <div class="participant"> between line 126, 127

Comment (cjqian) [reply frontend/src/components/stages/election_view.ts:128]: Done, thank you!

**Pull request merged**